libflux/src/lib.rs

```diff
@@ -1,3 +1,5 @@
+#![cfg_attr(feature = "strict", deny(warnings))]
```

Comment:
what does this do?

Reply:
This converts warnings into errors if the strict feature is enabled. This prevents us from enabling warnings for all builds and allows us to only have this enabled for CI. The Makefile also uses this mode by default.